front/src/views/user/user-settings/user-settings.view.js

```diff
@@ -0,0 +1,73 @@
+import { MmtTextInput, MmtSelect, MmtSubmitButton } from '@components';
+import axios from 'axios';
+const { serverPath, emailRegex } = require('@/config/config.dev.json');
+
+export default {
+  components: {
+    MmtTextInput,
+    MmtSelect,
+    MmtSubmitButton
+  },
+  data () {
+    return {
+      genders: [],
+      languages: [],
+      countries: [],
+      lastName: '',
+      firstName: '',
+      email: '',
+      gender: '',
+      language: '',
+      country: '',
+      isLoaded: false,
+      emailAlreadyTaken: false,
+      formValidity: true,
+      emailRules: [
+        email => RegExp(emailRegex).test(email)
+        || this.$t('userSettings.email.errors.regexNotMatch')
+      ]
+    };
+  },
+  methods: {
+    applySettings () {
+      axios.put(`${serverPath}api/userSettings`,
+        {
+          lastName: this.lastName,
+          firstName: this.firstName,
+          email: this.email,
+          gender: this.gender,
+          language: this.language,
+          country: this.country
+        })
+        .then(() => console.log('Settings updated'))
+        .catch(error => {
+          this.emailAlreadyTaken = error.response.data === 'Email already taken';
+        });
+    },
+    fillLoadedFields (data) {
+      this.genders = this.convertToSelectCompatible(data.genders, 'gender');
+      this.languages = this.convertToSelectCompatible(data.languages, 'language');
+      this.countries = this.convertToSelectCompatible(data.countries, 'country');
```

Comment:
it is vocabularies that are used not only in your component, and each time front will request the same data

Reply:
also I can't understand why you need converters